### PR TITLE
ControlBoardWrapper2 cleanup and small improvment

### PR DESCRIPTION
**Pull request by barbalberto**
removed unused variable
propagate device name to ControlBoardWrapper2 subdevices to have more meaningful error messages

### CODE REVIEW

File: src/libYARP_dev/include/yarp/dev/ImplementTorqueControl.h [hunk 1]
```diff
@@ -31,7 +31,7 @@ class YARP_dev_API yarp::dev::ImplementTorqueControl: public ITorqueControl
     double *temp;
     double *temp2;
     int    *temp_int;
-    yarp::dev::Pid *fake;
+//     yarp::dev::Pid *fake;
```
Comment (traversaro) [inline src/libYARP_dev/include/yarp/dev/ImplementTorqueControl.h]: +1 for just delete it, if we ever need this git remembers .

**Comment by barbalberto** [comment]
Test passed, can I merge this PR?
@traversaro @drdanz @randaz81

**Comment by traversaro** [comment]
@barbalberto Did you see the comment in https://github.com/robotology/yarp/pull/988#discussion_r90504009 (not really important, we can merge anyway, but it would be nice to get rid of the useless comment)?

**Comment by barbalberto** [comment]
ok, di it. Any other things?

**Comment by traversaro** [comment]
Ok for me!

**Pull request merged**